### PR TITLE
[Service] Pihole service

### PR DESCRIPTION
**Pull request by VonOx**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [ ] If your changes affects code, did your write the tests?
- [ ] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [ ] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [X] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

image image

### CODE REVIEW

**Comment by VonOx** [comment]
Ok guys ( @GladysAssistant/core ) , I think this PR need a first review.

This is my first integration, nothing fancy here, setup ip of your pihole server in configuration page and add box in your dashboard.

Any feedback are very welcome ( I'm still learning a lot in JS )
If any of you have an idea of how to grab the last percent on coverage, tell me :)

PS: You can run pihole in a docker container for testing https://hub.docker.com/r/pihole/pihole

File: front/src/config/i18n/fr.json [hunk 1]
```diff
@@ -838,6 +855,16 @@
         "defaultDeletionError": "Une erreur s'est produite lors de la suppression de l'appareil.",
         "conflictError": "L'appareil actuel est déjà dans Gladys."
       }
+    },
+    "pihole": {
+      "title": "Pi-hole",
+      "description": "Parlez à Gladys grâce à Telegram.",
+      "introduction": "INTROoooooooo",
```
Comment (atrovato) [inline front/src/config/i18n/fr.json]: Euh... not OK with INTROoooooooo

File: server/services/pihole/api/pihole.controller.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+module.exports = function PiholeController(piholeHandler) {
+  /**
+   * @api {get} /api/v1/pihole/getPiholeSummary
+   * @apiName getPiholeSummary
+   * @apiGroup PiHole
+   * @apiSuccessExample {json} Success-Example
+   */
+  async function getPiholeSummary(req, res) {
+    const result = await piholeHandler.getPiholeSummary(req.user.id);
+    res.json(result);
```
Comment (atrovato) [inline server/services/pihole/api/pihole.controller.js:12]: Do you think you can add test on it?

File: server/test/services/pihole/index.test.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  it('should faild to start', async () => {
+    const gladys = {
+      variable: {
+        getValue: fake.returns(null),
+      },
+    };
+    const piholeService = PiholeService(gladys, '5614b98c-6149-4db4-8ef5-a4c4281a9b7d');
+    try {
+      await piholeService.start();
+    } catch (e) {
```
Comment (atrovato) [inline server/test/services/pihole/index.test.js:58]: Please add `assert.fail()` before catch to ensure we need an error.

File: server/test/services/pihole/index.test.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+});
+
+describe('Pihole Service API', () => {
+  const controller = PiholeController(piholeHandler);
+  it('get /api/v1/pihole/getPiholeSummary', async () => {
+    const req = {};
+    const res = {
+      json: fake.returns(FakePiholeSummary),
+    };
+    await controller['get /api/v1/pihole/getPiholeSummary'].controller(req, res);
```
Comment (atrovato) [inline server/test/services/pihole/index.test.js:81]: Can you add assertion here?

File: server/utils/constants.js [hunk 5]
```diff
@@ -478,6 +478,7 @@ const DASHBOARD_BOX_TYPE = {
   HUMIDITY_IN_ROOM: 'humidity-in-room',
   USER_PRESENCE: 'user-presence',
   CAMERA: 'camera',
+  PIHOLE: 'pihole',
```
Comment (atrovato) [inline server/utils/constants.js:481]: Do we really an "pihole" box or a "network" box using pihole service?

**Review by atrovato — COMMENTED** [review]
Next step can be to install PiHole from Gladys (has mqtt broker).

**Comment by atrovato** [comment]
I think you forgot some descriptions:

**Comment by atrovato** [comment]
And intergation page talks about telegram key:


And once I click on save, it stay on "loading" page.
I set URL with a final slash ("http://192.168.1.11/") and it fails, it tries to call "http://http://192.168.1.11//admin..." with double http:// and //.
You can use https://nodejs.org/api/url.html to build proper URLs.

**Comment by atrovato** [comment]
Now I'm ok, but I have some weird data on box:


No need to explicitly say Pihole status, status is enough.
For more, I'm in french, and I see "enabled", we may need to overwritte it.

And is there an auto-refresh? I disabled pi-hole and I had to "refresh" or change page to see status changed.

File: front/src/config/i18n/fr.json (hunk 1 above)
Comment (VonOx) [reply front/src/config/i18n/fr.json]: Fixed :D

**Pull request closed without merging**